internal/lobby/lobby.go

```diff
@@ -64,3 +65,23 @@ func CreateLobby(options LobbyOptions) *Lobby {
 		Players:         make(map[ClientID]*Player),
 	}
 }
+
+//returns top 3 players in a lobby
+func (l *Lobby) GetLeaderboard() []*Player {
+	l.mu.Lock()
+	defer l.mu.Unlock()
+
+	players := make([]*Player, 0, len(l.Players))
+	for _, player := range l.Players {
+		players = append(players, player)
+	}
+
+	sort.Slice(players, func(i, j int) bool {
+		return players[i].Score > players[j].Score
+	})
+
+	if len(players) > 3 {
+		return players[:3]
+	}
+	return players
+}
```

Comment:
Lobbies are active games, they are not permanent.
The results of the game will be saved inside the common.Game struct located in "internal/common/types".
So move/adapt the methods from Lobby to Game.